### ISSUE TITLE
Upgrade Android Library to Cast v3

### ISSUE THREAD

**Issue by charliesbox** [react-native-google-cast/react-native-google-cast#35]
This might mean that we could get rid of the companion library, as the last one influenced the new version of the cast sdk.

https://developers.google.com/cast/v2/ccl_migrate_sender

**Comment by charliesbox** [comment]
dsfsdsfs

**Comment by deepmehtait** [comment]
@charliesbox : Upgrade is not direct. Concepts of start scan and stop scan with the list of devices are not available in V3. Any idea how we can convert CCL to V3?

**Comment by developersoul** [comment]
Now the chromecast button handle on sdk v3, so to make this migration it's necessary another approach for example instead of methods it's necessary an UI component to handle the state of the devices available and connect and disconnect...

**Comment by seftonmonk** [comment]
Anybody go this library working with v3?

**Comment by charliesbox** [comment]
@seftonmonk I'm gonna start working on that this weekend. I didn't have too much progress before cause most of my issues came from Android, and my testing using the emulator didn't give shed me lights about it.

However, I got a Pixel recently and having both devices will let me focus better on the new architecture.

**Comment by seftonmonk** [comment]
That's great news! Thanks for the update. Looking forward to it, great work in the last version.

**Comment by seftonmonk** [comment]
@charliesbox any success? sorry to bother ;)

**Comment by nehacurefit** [comment]
Hi @charliesbox , Can u please provide update on cast v3 integration, I tried it but many of the handles are not there in v3 like connect to device, getting devices added, removed callback.

**Comment by cobarx** [comment]
@charliesbox By any chance do you know if it's possible to detect devices with the v3 SDK? Our team is currently working on this and I'm not sure if equivalent functionality to the CCL even exists. I'm going to try the way they write it up on the Google Cast site for a first pass.

**Comment by petrbela** [comment]
We've rewritten the library with v3, check out https://github.com/flextv/react-native-google-cast. We've been using it in production for two months now.

**Comment by cobarx** [comment]
@petrbela We ran into a lot of issues with v2 where on Android it worked with certain devices but not others. Given that Google has abandoned the CCL, the only option was to move to v3. I had started working on it, but it's very exciting to see that you've already figured it out and gotten it working. There were a number of questions I had that you resolved like how to create a compatible ReactNative activity and whether we should implement our button or just render the Google Cast provided one unmodified.

Looking forward to trying this out and will send PR's if there's anything our team can add.
